h5netcdf/tests/test_h5netcdf.py

```diff
@@ -737,3 +737,12 @@ def test_reading_unlimited_dimensions_created_with_c_api(tmp_netcdf):
         assert f['dummy2'].shape == (3, 0, 0)
         f.groups['test']['dummy3'].shape == (3, 3)
         f.groups['test']['dummy4'].shape == (0, 0)
+
+
+def test_reading_unused_unlimited_dimension(tmp_netcdf):
+    """Test reading a file with unused dimension of unlimited size"""
+    with h5netcdf.File(tmp_netcdf, 'w') as f:
+        f.dimensions = {'x': None}
+        f.resize_dimension('x', 5)
+
+    f = h5netcdf.File(tmp_netcdf, 'r')
```

Comment:
Thanks for adding the test. For completeness, could you also add a check that dimensions are as expected? e.g., `assert f.dimensions == {'x': None}`.